### ISSUE TITLE
Nested tuples in a for directive return a compilation error: "expected closing tag for "li""

### ISSUE THREAD

**Issue by Malian** [msaraiva/surface#185]
Hi here,

In the project I am working on, I have an array composed of nested struct
```elixir
my_array = [{:a, {"vala1, "vala2"}}, {:b, {"valb1", "valb2"}}]
```
I would like to loop over each element in my component so I wrote the following

```elixir
<ul>
  <li :for={{ {_, {val1, val2}} <- my_array }}>
    {{ val1 }} - {{val2}}
  </li>
</ul>
```
But it results in a compilation error that says: "expected closing tag for li".

Did I miss something?

If it's a bug, I would be glad to help to fix it but I need some guidance.

Please find the complete example below and a working (in fact a "non working" 😉) sample app [here](https://github.com/Malian/SurfaceNestedTupleBug).
```elixir
defmodule NestedTuple do
  use Surface.Component

  def render(assigns) do
    ~H"""
    <ul>
      <li :for={{ {a, {b, c}} <- [{"a", {"b", "c"}}]}}>
        {{ c }}
      </li>
    </ul>
    """
  end
end
```

```console
Erlang/OTP 23 [erts-11.1.1] [source] [64-bit] [smp:12:12] [ds:12:12:10] [async-threads:1] [hipe]

Compiling 13 files (.ex)

== Compilation error in file lib/surface_nested_tuple_bug_web/live/components/nested_tuple.ex ==
** (Surface.Compiler.ParseError) lib/surface_nested_tuple_bug_web/live/components/nested_tuple.ex:7: expected closing tag for "li"
    (surface 0.1.0-rc.1) lib/surface/compiler.ex:109: Surface.Compiler.compile/5
    (surface 0.1.0-rc.1) expanding macro: Surface.sigil_H/2
    lib/surface_nested_tuple_bug_web/live/components/nested_tuple.ex:5: NestedTuple.render/1
    (elixir 1.11.0) lib/kernel/parallel_compiler.ex:314: anonymous fn/4 in Kernel.ParallelCompiler.spawn_workers/7
```

**Comment by msaraiva** [comment]
Hi @Malian!

This should have been fixed in #171 but I can confirm the problem using the latest version on master. 

Replacing

```
<li :for={{ {a, {b, c}} <- [{"a", {"b", "c"}}]}}>
```
with
```
<li :for={{ {a, {b, c}} <- [{"a", {"b", "c"} }]}}>
                                           ^^^
```

fixes the compilation error so I believe we missed some edge cases.

/cc @lnr0626

**Comment by lnr0626** [comment]
definitely an oversight, should have a fix up shortly

**Comment by lnr0626** [comment]
Hi @Malian!

This should be fixed in master (as of b745e13), thanks for the test case - it helped to pinpoint the issue quickly :-)
